server/container_restore.go

```diff
@@ -75,6 +75,23 @@ func (s *Server) CRImportCheckpoint(
 
 	var restoreArchivePath string
 	if restoreStorageImageID != nil {
+		sb, err := s.getPodSandboxFromRequest(ctx, sbID) // Note that we might call getPodSandboxFromRequest with a different sbID later. Is that necessary?
```

Comment:
Previous discussion about `sbID`: https://github.com/cri-o/cri-o/pull/8590#discussion_r1754545705

>  i don't think we know for sure what's the sandbox ID yet. 
> …
> The code to restore into an existing sandbox ID is a left-over from the initial pull request which supported to restore a pod. This is no longer possible with the current code. Most of the code was removed. This should haven been also removed.

Reply:
@adrianreber If I understand this correctly, the sandbox ID logic can be simplified with the two commits I have just added to this PR:
- Remove the option to load a sandbox ID from the snapshot image
- Centralize handling of `CreateContainerRequest.PodSandboxId`